java-frontend/src/main/java/org/sonar/java/se/xproc/MethodYield.java

```diff
@@ -151,27 +129,31 @@ private boolean applicableOnVarArgs(int index, List<Type> types) {
     return argumentType.isArray() || argumentType.is("<nulltype>");
   }
 
-  private static Set<ProgramState> programStatesForConstraint(Collection<ProgramState> states, SymbolicValue invokedArg, Constraint constraint) {
-    Set<ProgramState> programStates = new LinkedHashSet<>();
-    if (constraint instanceof ObjectConstraint) {
-      ObjectConstraint objectConstraint = (ObjectConstraint) constraint;
-      states.forEach(state -> programStates.addAll(invokedArg.setConstraint(state, objectConstraint)));
-    } else if (constraint instanceof BooleanConstraint) {
-      BooleanConstraint booleanConstraint = (BooleanConstraint) constraint;
-      states.forEach(state -> programStates.addAll(invokedArg.setConstraint(state, booleanConstraint)));
-    }
-    return programStates;
+  public Constraint[] parametersConstraints() {
+    return parametersConstraints;
+  }
+
+  @CheckForNull
+  public Constraint parameterConstraint(int parameterIndex) {
+    return parametersConstraints[parameterIndex];
+  }
+
+  public void setParameterConstraint(int index, @Nullable Constraint constraint) {
+    Preconditions.checkArgument(index < parametersConstraints.length);
+    parametersConstraints[index] = constraint;
+  }
+
+  public boolean generatedByCheck(SECheck check) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "check". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)